R/diagnostic_test.R

```diff
@@ -0,0 +1,198 @@
+#' Diagnostic testing
+#'
+#' @description
+#' Run a set of diagnostic tests to check for common issues found when setting
+#' up R on a DfE
+#' system. This includes:
+#'   - Checking for proxy settings in the Git configuration
+#'   - Checking for correct download method used by renv (curl)
+#'
+#' @inheritParams check_proxy_settings
+#'
+#' @return List object of detected parameters
+#' @export
+#'
+#' @examples
+#' diagnostic_test()
+diagnostic_test <- function(
+    clean = FALSE,
+    verbose = FALSE) {
+  results <- c(
+    check_proxy_settings(clean = clean, verbose = verbose),
+    check_github_pat(clean = clean, verbose = verbose),
+    check_renv_download_method(clean = clean, verbose = verbose)
```

Comment:
Could we add an additional check for http.sslVerify in their Git config too?

`system("git config --global http.sslVerify false")` - was a line in the original proxy script and we should be discouraging analysts from having it in their configs anymore (as it's not longer necessary)